core/src/main/java/org/mule/util/ArrayUtils.java

```diff
@@ -138,5 +141,57 @@ else if (componentType.equals(Double.TYPE))
         result.retainAll(Arrays.asList(b));
         return result.toArray(new String[result.size()]);
     }
-    
+
+    public static int getLength(Object array)
+    {
+        if (array == null)
+        {
+            return 0;
+        }
+
+        checkArgument(array.getClass().isArray(), String.format("Object of type %s is not an array", array.getClass().getName()));
+
+        Class<?> componentType = array.getClass().getComponentType();
```

Comment:
I don't understand how that method works, thus I don't want to repackage it Mule. I don't want to depend on that library either. Thanks for the tip though, I thought it was not possible to do that in Java